lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -279,6 +282,13 @@ public void submit() throws IOException, YarnException {
     }
 
     this.yarnClient.submitApplication(applicationSubmissionContext);
+
+    if (this.isUnmanaged) {
+      // For Unmanaged AM mode, add a new app token to the
+      // current process so it can talk to the RM as an AM.
+      final Token<AMRMTokenIdentifier> token = this.yarnClient.getAMRMToken(this.applicationId);
+      this.tokenProvider.addTokens(UserCredentialSecurityTokenProvider.serializeToken(token));
```

Comment:
OK, so we are getting the driver token from RM to set it at client side so that client will have the same permission as driver. 

If driver runs somewhere else, does RM set this token to the driver machine?